File: T34-Tuplespaces-2025/Client-Java/src/main/java/pt/ulisboa/tecnico/tuplespaces/client/ClientMain.java
package pt.ulisboa.tecnico.tuplespaces.client;

import pt.ulisboa.tecnico.tuplespaces.client.grpc.ClientService;
import static pt.ulisboa.tecnico.tuplespaces.client.ClientDebug.debug;

public class ClientMain {

    public static void main(String[] args) {

        System.out.println(ClientMain.class.getSimpleName());

        if (3 < args.length || args.length < 2) {
            System.err.println("Bad Argument(s)!");
            System.err.println("Usage: mvn exec:java -Dexec.args=<server.host>:<server.port> <client_id> [-debug]");
            return;
        }

        if (args.length==3 && args[2].equals("-debug"))
            ClientDebug.setDebugFlag(true);

        // receive and print arguments
        debug(String.format("Received %d arguments%n", args.length));
        for (int i = 0; i < args.length; i++) {
             debug(String.format("arg[%d] = %s%n", i, args[i]));
        }
        // get the host and the port of the server or front-end
        final String host_port = args[0];
        final int client_id = Integer.parseInt(args[1]) ;     

        CommandProcessor parser = new CommandProcessor(new ClientService(host_port, client_id));
        parser.parseInput();

    }
}

File: T34-Tuplespaces-2025/Client-Python/client_main.py
import sys
from typing import List
from client_service import ClientService
from command_processor import CommandProcessor
from client_debug import ClientDebug
from client_debug import ClientDebug


class ClientMain:
    @staticmethod
    def main(args: List[str]):
        print("ClientMain\n")

        if len(args) == 3 and args[2] == "-debug":
            ClientDebug._debug_flag = True  # Ativa a flag de debug

        # receive and print arguments
        ClientDebug.debug(f"Received {len(args)} arguments")
        for i, arg in enumerate(args):
            ClientDebug.debug(f"arg[{i}] = {arg}\n")

        # check arguments
        if len(args) < 2:
            ClientDebug.debug("Argument(s) missing!", file=sys.stderr)
            ClientDebug.debug("Usage: python3 client_main.py <server.host>:<server.port> <client_id> [-debug]", file=sys.stderr)
            return

        

        host_port = args[0]
        client_id = args[1]


        service = ClientService(host_port, client_id)
        parser = CommandProcessor(service)
        parser.parse_input()
        service.shutdown()

if __name__ == "__main__":
    ClientMain.main(sys.argv[1:])